src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -1686,10 +1687,21 @@ ssize_t sockinfo_tcp::rx(const rx_call_t call_type, iovec* p_iov, ssize_t sz_iov
 				ret = 0;
 			}
 			goto err;
-        	}
-        	ret = rx_wait(poll_count, block_this_run);
-        	if (unlikely(ret < 0)) goto err;
+		}
+
+		if (m_timer_pending) {
```

Comment:
As we discussed F2F, I suggest replacing all of this with:

```
inline int sockinfo_tcp::rx_wait_unlocked(int &poll_count, bool is_blocking)
{
	if (m_timer_pending) {
		m_tcp_con_lock.lock();
		tcp_timer();
		unlock_tcp_con(is_locked);
	}

        return rx_wait_helper(poll_count, is_blocking);
}
```
The above should go immediately following the `sockinfo_tcp::rx_wait()` implementation.
And then here you should do:

```
        	ret = rx_wait_unlocked(poll_count, block_this_run);
        	if (unlikely(ret < 0)) goto err;

```